env0/data_sshkey_test.go

```diff
@@ -0,0 +1,67 @@
+package env0
+
+import (
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"testing"
+)
+
+func TestUnitSshKeyDataSourceById(t *testing.T) {
+	resourceType := "env0_ssh_key"
+	resourceName := "test"
+	resourceFullName := dataSourceAccessor(resourceType, resourceName)
+	sshKey := client.SshKey{
+		Id:    "id0",
+		Name:  "name0",
+		Value: "Key🔑",
+	}
+
+	testCase := resource.TestCase{
+		ProviderFactories: testUnitProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: dataSourceConfigCreate(resourceType, resourceName, map[string]string{
+					"id": sshKey.Id,
+				}),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceFullName, "id", sshKey.Id),
+					resource.TestCheckResourceAttr(resourceFullName, "name", sshKey.Name),
+				),
+			},
+		},
+	}
+
+	runUnitTest(t, testCase, func(mock *client.MockApiClientInterface) {
+		mock.EXPECT().SshKeys().AnyTimes().Return([]client.SshKey{sshKey}, nil)
```

Comment:
Can we try to be more specific with `.Times(1)`?
This is also relevant for all other tests

Reply:
20f7ef7
worked only for resource tests file. data must have `AnyTimes` from some reason

Reply:
Can you tell maybe how many times it runs? (you can check out using debugger). 
I'm asking because it's important to understand this tho.If it's only two - we can add a setting that prevents the second time from running so we'd be able to assert the number of calls easily.

Reply:
where do you see it on the debugger? what line

Reply:
🤙

Reply:
a3bca40